riscv_isac/plugins/internaldecoder.py

```diff
@@ -1210,7 +1211,10 @@ def arith_ops(self, instrObj):
                 instrObj.rs2 = rs2
                 instrObj.rd = rd
             elif funct7 == 0b0000100:
-                if rs2[0] == 0b0:
+# pack and zext.h have same opcode, func3, funct7 only diffrence is in rs2 value
+# for zext.h rs2 is always 0, if pack instruction is used with x0 as rs2
+# then cannot distinguish from each other, hence using cover label to differentiate.
+                if rs2[0] == 0b0 and  "pack" not in self.labels:

```

Comment:
This is not the right way to deal with pseudo ops. This will cause problems when the input labels contain both `pack` and `zext.h`. Pseudo-ops need some further thought. There are 2 ways to handle this. The first is to statically encode the pseudo-ops in the decoder and add another field in the instruction object which contains the list of pseudo ops satisfied by that instruction instance. The second is to express the coverpoints for the pseudo ops as the original ops and add another node in the `covergroup` node called `p_op_cond:`. So the condition to check whether a particular instruction should be considered for a covergroup also checks if the `p_op_cond` is satisfied before updating the coverpoints. In this case `zext.h` covergroup becomes:
```
zext.h:
    config: 
      - check ISA:=regex(.*I.*B.*)
      - check ISA:=regex(.*I.*Zbb.*)
    opcode: 
      pack: 0
      packw: 0
    p_op_cond: rs2 == x0
    rs1: 
      <<: *all_regs
    rd: 
      <<: *all_regs
    op_comb: 
      <<: *ifmt_op_comb
    val_comb:
      'rs1_val == 0': 0
      'rs1_val == 0x80': 0
      'rs1_val == 0xFF80': 0
      abstract_comb:
        'walking_ones("rs1_val", xlen, False)': 0
        'walking_zeros("rs1_val", xlen, False)': 0
        'uniform_random(20, 100, ["rs1_val"], [xlen])': 0
``` 

Thoughts? @allenjbaum @neelgala